src/redux/user/userSlice.js

```diff
@@ -0,0 +1,123 @@
+import { createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { toast } from 'react-toastify';
+
+const BASE_URL = 'http://localhost:3000/';
+
+const tokenKey = 'auth_token';
+
+const initialToken = localStorage.getItem(tokenKey) || null;
+
+const initialState = {
+  auth_token: initialToken,
+  isLoggedIn: false,
+  user: {
+    id: null,
+    name: null,
+    email: null,
+    password: null,
+  },
```

Comment:
Like I mentioned on another [PR](https://github.com/leloufadel/full-stuck-project/pull/26#pullrequestreview-1839581752), I am unsure why we are still using password if you've made changes on the backend to no longer use passwords.